Disabled adding to other user's space in pipelines component and set the recent-pipeline's build config

chrislessard opened the pull request:
These patches accomplish two things:

- The first disables buttons that allow a user to modify another's space. This addresses https://github.com/openshiftio/openshift.io/issues/3386 

- Adds a default value to the build-config in the recent-pipelines component

chrislessard commented:
Here's a screenshot, where I've visited one of jkang@redhat.com's spaces.
![2018-04-30-171244_1916x1026_scrot](https://user-images.githubusercontent.com/17230733/39450559-c5f5766c-4c99-11e8-9ec0-53d36904a035.png)

christianvogt commented:
My general rule is that a button should only be disabled if the user has a way to enable it or if the disabled state also presents the user with additional info regarding the disabled state (eg. a popover). Otherwise the element should not appear at all as it confuses the user, why they cannot perform the action.

@catrobson what guidelines does your team suggest?

joshuawilson commented:
I would agree, only disable what can be enabled. 

It should be clear that you are in someone else's space.

chrislessard commented:
@christianvogt @joshuawilson @catrobson Cool, that makes sense to me. I'll make sure the "plus" button is removed if you're in another user's space.

chrislessard commented:
@christianvogt @joshuawilson @catrobson I suppose I should have included a screenshot for this, but the "Add to space" button does show text when the user hover's over it. The message will inform them that they cannot add to another user's space. I can't add this to an icon, however, so I'll make sure the "plus" button is removed in this instance.

<img width="1440" alt="screen shot 2018-05-01 at 11 36 43 am" src="https://user-images.githubusercontent.com/17230733/39480001-eda89f22-4d34-11e8-8a2e-801cb58c7455.png">

alien-ike commented:
### Ike Plugins (test-keeper)

Thank you @chrislessard for this contribution!

<img align="left" src="https://cdn.rawgit.com/bartoszmajsak/ike-prow-plugins/2025328b70bd1879520411b3cacadee61a49641a/docs/images/arquillian_ui_failure_128px.png">

It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


For more information please head over to official [documentation](http://arquillian.org/ike-prow-plugins/#_test_keeper_plugin). You can find there how to configure the plugin - for example exclude certain file types so if PR contains only them it won't be checked.

andrewazores commented on src/app/dashboard-widgets/pipelines-widget/pipelines-widget.component.spec.ts:
```diff
@@ -0,0 +1,153 @@
+  let mockRouterEvent: any = {
+    'id': 1,
+    'url': 'mock-url'
+  };
```
Can this be moved into the factory function? It doesn't appear that you reference it anywhere below in the test cases, only in the Router factory.

andrewazores commented on src/app/dashboard-widgets/pipelines-widget/pipelines-widget.component.spec.ts:
```diff
@@ -0,0 +1,153 @@
+
+  let mockRouter;
+  let mockActivatedRoute;
+  let mockLocationStrategy;
```
mockRouter can be inlined inside the factory function, and the other two are unused?

Same for the ones above, ex. `authentication` - if you don't need a reference to them in the test cases, please don't hold a reference to them at the top here. It's just extra cruft to look at that does nothing.

andrewazores commented on src/app/dashboard-widgets/pipelines-widget/pipelines-widget.component.spec.ts:
```diff
@@ -0,0 +1,153 @@
+  });
+
+  it('should recognize that the user owns the space', function(this: TestingContext) {
+    const userService: jasmine.SpyObj<UserService> = TestBed.get(UserService);
```
unused

andrewazores commented on src/app/dashboard-widgets/pipelines-widget/pipelines-widget.component.spec.ts:
```diff
@@ -0,0 +1,153 @@
+  });
+
+  it('should recognize that the user does not own the space', function(this: TestingContext) {
+    const userService: jasmine.SpyObj<UserService> = TestBed.get(UserService);
```
unused

andrewazores commented on src/app/dashboard-widgets/pipelines-widget/pipelines-widget.component.ts:
```diff
@@ -26,28 +29,52 @@ import { PipelinesService } from '../../space/create/pipelines/services/pipeline
+      .subscribe(length => setTimeout(() => this.buildConfigsCount = length))
+    );
+
+    this.context.current.subscribe((ctx: Context) => {
```
Add to subscriptions array

andrewazores commented on src/app/dashboard-widgets/pipelines-widget/pipelines-widget.module.ts:
```diff
@@ -15,6 +19,7 @@ import { PipelinesWidgetComponent } from './pipelines-widget.component';
   declarations: [PipelinesWidgetComponent],
-  exports: [PipelinesWidgetComponent]
+  exports: [PipelinesWidgetComponent],
+  providers: [ PipelinesService ]
```
Why did this need to be added here? The component was already using the PipelinesService before, wasn't it? In fact it appears to be providing its own PipelinesService. Should that have been removed?

Also, whitespace is inconsistent.

chrislessard replied:
I moved it here to make some tests work, because I was getting a complaint about the PipelinesService not being provided even though I had a mock in place. I've removed it from the component.

andrewazores replied:
Whitespace within the square brackets is still inconsistent

andrewazores commented on src/app/dashboard-widgets/pipelines-widget/pipelines-widget.component.ts:
```diff
@@ -26,28 +29,52 @@ import { PipelinesService } from '../../space/create/pipelines/services/pipeline
+    });
+
+    this.subscriptions.push(this.userService.loggedInUser.subscribe((usr: User) => {
+      this.loggedInUser = usr;
```
`usr`? Maybe `user` is better?

chrislessard commented:
@andrewazores I've changed the tests to reflect that the space ownership is now being checked in the parent component, and I've cleaned up the bits you pointed out.

andrewazores commented on src/app/dashboard-widgets/pipelines-widget/pipelines-widget.component.spec.ts:
```diff
@@ -25,53 +25,42 @@ import { PipelinesService } from '../../space/create/pipelines/services/pipeline
 class HostComponent { }
 
-describe('PipelinesWidgetComponent', () => {
+fdescribe('PipelinesWidgetComponent', () => {
```
fdescribe

fabric8cd commented:
@chrislessard fabric8/fabric8-ui:SNAPSHOT-PR-2842-3 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2842-fabric8-ui.badger.fabric8.io

fabric8cd commented:
@chrislessard fabric8/fabric8-ui:SNAPSHOT-PR-2842-4 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2842-fabric8-ui.badger.fabric8.io

fabric8cd commented:
@chrislessard fabric8/fabric8-ui:SNAPSHOT-PR-2842-6 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2842-fabric8-ui.badger.fabric8.io

catrobson commented:
+1 that we should remove, not disable, these buttons.  It looks like on the current dashboard implementation we would need to disable:  Add to Space, Import Codebase (as this should be going to wizard/launcher moving forward), and Add Codebase (this should also be going to wizard/launcher moving forward) in the case that you are looking at a space you do not own.

fabric8cd commented:
@chrislessard fabric8/fabric8-ui:SNAPSHOT-PR-2842-7 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2842-fabric8-ui.badger.fabric8.io

fabric8cd commented:
@chrislessard fabric8/fabric8-ui:SNAPSHOT-PR-2842-14 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2842-fabric8-ui.badger.fabric8.io

fabric8cd commented:
@chrislessard fabric8/fabric8-ui:SNAPSHOT-PR-2842-15 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2842-fabric8-ui.badger.fabric8.io

dlabrecq commented on src/app/dashboard-widgets/pipelines-widget/pipelines-widget.component.html:
```diff
@@ -17,7 +19,10 @@ <h3>No Pipelines have run</h3>
       </p>
       <div class="f8-blank-slate-main-action">
-        <button id="spacehome-pipelines-add-button" class="btn btn-primary btn-lg" (click)="addToSpace.emit()">Add to Space</button>
+      <div *ngIf="userOwnsSpace">
```
PR https://github.com/fabric8-ui/fabric8-ui/pull/2898 uses ng-container instead of a div tag. I believe we should be consistent and use ng-container here. The extra div tags add to the DOM structure, which is unnecessary.

chrislessard replied:
Good catch.

fabric8cd commented:
@chrislessard fabric8/fabric8-ui:SNAPSHOT-PR-2842-18 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2842-fabric8-ui.badger.fabric8.io

dlabrecq commented:
No further changes necessary.

fabric8cd commented:
@chrislessard fabric8/fabric8-ui:SNAPSHOT-PR-2842-20 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2842-fabric8-ui.badger.fabric8.io

fabric8cd commented:
@chrislessard fabric8/fabric8-ui:SNAPSHOT-PR-2842-21 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2842-fabric8-ui.badger.fabric8.io
